### DISCUSSION TITLE
Error on setting nodes in 12.3.1 React

### DISCUSSION THREAD

**Discussion by Djuppi** [xyflow/xyflow#4703] [General]
I have a flow-diagram that I have been working on @xyflow/react v12.1.1, and everything worked completely fine. Then I thought maybe I would update to the latest version, but that crashed everything... I get this error message when I try to setNodes the way I did before (see error picture and code snippet
<img width="1251" alt="Screenshot 2024-10-02 at 09 48 13" src="https://github.com/user-attachments/assets/dffca099-3e42-42e1-8dd7-4a020a2a82df">
. 

![image](https://github.com/user-attachments/assets/631bf262-57e3-4bce-a2b4-86200ae0355c)

Im not using any Store like Zustan, cause I'm saving to our database.

Does anyone else get this issue?

**Comment by Djuppi** [comment]
I Investigated with previous versions, and it seems to happen from v12.3.0 and forward.

**Comment by moklick** [comment]
Since this is a bug, we will continue here: https://github.com/xyflow/xyflow/issues/4705
